apisix/core/config_etcd.lua

```diff
@@ -257,6 +260,33 @@ local function run_watch(premature)
 end
 
 
+local function run_watch(premature)
+    local run_watch_th = ngx_thread_spawn(do_run_watch, premature)
+
+    ::restart::
+    local check_worker_th = ngx_thread_spawn(function ()
+        while not exiting() do
+            ngx_sleep(0.1)
+        end
+    end)
+
+    local ok, err = ngx_thread_wait(check_worker_th)
```

Comment:
I think you should monitor the `run_watch_th` too in case that the `run_watch_th` fails

Reply:
The `do_run_watch` function will not return, it has an internal retry mechanism.  The origin logic did not handle this so I didn't do this either.

Reply:
We should handle the situation that the `run_watch_th` thread crashes

Reply:
Do we need restart `run_watch_th` when it fails?

Reply:
I think the code may like this
```
   ::restart::
    local run_watch_th = ngx_thread_spawn(do_run_watch, premature)
    local check_worker_th = ngx_thread_spawn(function ()
        while not exiting() do
            ngx_sleep(0.1)
        end
    end)

    local ok, err = ngx_thread_wait(run_watch_th, check_worker_th)
    if not ok then
        log.error("wait worker checker failed, retart checker, error: " .. err)
        ngx_thread_kill(run_watch_th)
        ngx_thread_kill(check_worker_th)
        goto restart
    end
```